### PR TITLE
Adding more discovery tests and additional changes to make them more …

### PR DESCRIPTION
**Pull request by jbeuree**
…reliable

## Description
Additional discovery UI tests and some reliability changes

## Jira link(s)
- https://opennms.atlassian.net/browse/HS-1839

## Checklist
* [ ] Follows Lōkahi's [development guidelines.](https://github.com/OpenNMS-Cloud/lokahi/wiki/Development-Guidelines)
* [ ] Appropriate reviewer(s) have been selected.
* [ ] Jira issue(s) have been updated to "In Review".
* [ ] Includes [appropriate tests.](https://github.com/OpenNMS-Cloud/lokahi/wiki/Test-Strategy)
* [ ] Documentation has been updated as necessary.
* [ ] Notify devops of changes to the Charts.
* [ ] Notify documentation team of any changes to names of screens or features (affects URLs).

### CODE REVIEW

File: test-automation/ui-tests/src/test/java/org/opennms/horizon/systemtests/steps/DiscoverySteps.java [hunk 1]
```diff
@@ -69,40 +71,184 @@ public class DiscoverySteps {
...
+    }
+
+    private String getContainerIPs(String nodeNames) {
+        String ips = "";
+        String[] nodeNameArray = nodeNames.split(",");
+        for (int i = 0; i < nodeNameArray.length; i++) {
+            String nodeName = nodeNameArray[i].trim();
+            GenericContainer container = nodes.get(nodeName);
+            assertNotNull("Must have a container for node " + nodeName, container);
+            if (! ips.isEmpty()) {
```
Comment (bouff) [inline test-automation/ui-tests/src/test/java/org/opennms/horizon/systemtests/steps/DiscoverySteps.java:164]: Nit: code style: extra space

File: test-automation/ui-tests/src/test/java/org/opennms/horizon/systemtests/steps/DiscoverySteps.java [hunk 2]
```diff
@@ -69,40 +71,184 @@ public class DiscoverySteps {
...
+        String ipList = getContainerIPs(nodeNames);
+
+        performDiscovery(discoveryName, LocationSteps.getLocationName(), port, community, ipList);
+
+    }
+
+    private String getContainerIPs(String nodeNames) {
+        String ips = "";
+        String[] nodeNameArray = nodeNames.split(",");
+        for (int i = 0; i < nodeNameArray.length; i++) {
```
Comment (bouff) [inline test-automation/ui-tests/src/test/java/org/opennms/horizon/systemtests/steps/DiscoverySteps.java:160]: Nit: code style:  use enhanced loop: `for (String nodeName : nodeNameArray)`

File: test-automation/ui-tests/src/test/java/org/opennms/horizon/systemtests/steps/DiscoverySteps.java [hunk 3]
```diff
@@ -69,40 +71,184 @@ public class DiscoverySteps {
...
+            String nodeName = nodeNameArray[i].trim();
+            GenericContainer container = nodes.get(nodeName);
+            assertNotNull("Must have a container for node " + nodeName, container);
+            if (! ips.isEmpty()) {
+                ips += ", " + getContainerIP(container);
+            } else {
+                ips += getContainerIP(container);
```
Comment (bouff) [inline test-automation/ui-tests/src/test/java/org/opennms/horizon/systemtests/steps/DiscoverySteps.java:167]: Not: code style: use `StringBuilder` for concat of `String` in a loop.

File: test-automation/ui-tests/src/test/java/org/opennms/horizon/systemtests/steps/DiscoverySteps.java [hunk 4]
```diff
@@ -69,40 +71,184 @@ public class DiscoverySteps {
...
+            }
+        }
+        return ips;
+    }
+
```
Comment (bouff) [inline test-automation/ui-tests/src/test/java/org/opennms/horizon/systemtests/steps/DiscoverySteps.java:172]: Nit: extra line

File: test-automation/ui-tests/src/test/java/org/opennms/horizon/systemtests/steps/DiscoverySteps.java [hunk 5]
```diff
@@ -69,40 +71,184 @@ public class DiscoverySteps {
...
+    private int convertStringIPToInt(String stringAddr) {
+        try {
+            InetAddress i = InetAddress.getByName(stringAddr);
+            return ByteBuffer.wrap(i.getAddress()).getInt();
+        } catch (UnknownHostException e) {
+            throw new RuntimeException("Test error converting string ip '" + stringAddr + "' to properly formatted inet addr");
+        }
+    }
+
```
Comment (bouff) [inline test-automation/ui-tests/src/test/java/org/opennms/horizon/systemtests/steps/DiscoverySteps.java:229]: Nit: extra lines

**Review by bouff — APPROVED** [review]
LGTM, only some minor (non-blocking) nits.

File: test-automation/ui-tests/src/test/java/org/opennms/horizon/systemtests/pages/InventoryPage.java [hunk 6]
```diff
@@ -45,42 +47,64 @@ public enum Status {
...
         return nodesList.findBy(Condition.text(nodeSysName)).isDisplayed();
     }
 
-    public static void verifyNodeStatus(Status status, String nodename) {
+    public static void verifyNodeStatus(Status status, String nodeManagementIp) {
+        LeftPanelPage.clickOnPanelSection("inventory");
+
         // The inventory page doesn't refresh on its own. Need to periodically check and force a refresh
-        String itemStatusSearch = "//ul[@class='cards']/li[//li[@data-test='management-ip']/span/text()='" + nodename + "']//div[@title='Status']//span[text()='" + status + "']";
+        String itemStatusSearch = "//ul[@class='cards']/li[.//li[@data-test='management-ip']/span/text()='" + nodeManagementIp + "']//div[@title='Status']//span[text()='" + status + "']";
```
[the remark this answers is not in the document]
Comment (jbeuree) [reply test-automation/ui-tests/src/test/java/org/opennms/horizon/systemtests/pages/InventoryPage.java:63]: Yeah, wanted to go through a bunch separately and look at putting in more IDs, probably next week

**Pull request merged**